docs(linter): Add config option docs to `jest/consistent-test-it` rule.

connorshea opened the pull request:
Part of #14743.

Generated docs:

```md
## Configuration

This rule accepts a configuration object with the following properties:

### fn

type: `"it" | "test"`

default: `"test"`

Decides whether to use `test` or `it`.

### withinDescribe

type: `"it" | "test"`

default: `"it"`

Decides whether to use `test` or `it` within a `describe` scope.
If only `fn` is provided, this will default to the value of `fn`.

```

camc314 commented on crates/oxc_linter/src/rules/jest/consistent_test_it.rs:
```diff
@@ -16,66 +18,51 @@ use crate::{
     IT,
     Test,
-    Xit,
-    Xtest,
```
was moving these different test methods deliberate? if so why?

camc314 replied:
i see, these options were actually supported. but only by us, not the original rule. lets remove the support

camc314 commented:
### Merge activity

* **Dec 2, 11:55 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 2, 11:58 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
docs(linter): Add config option docs to `jest/consistent-test-it` rule. (#16286)

Part of #14743.

Generated docs:

```md
## Configuration

This rule accepts a configuration object with the following properties:

### fn

type: `"it" | "test"`

default: `"test"`

Decides whether to use `test` or `it`.

### withinDescribe

type: `"it" | "test"`

default: `"it"`

Decides whether to use `test` or `it` within a `describe` scope.
If only `fn` is provided, this will default to the value of `fn`.

```
